internal/impl/aws/input_kinesis.go

```diff
@@ -593,53 +593,9 @@ func (k *kinesisReader) runBalancedShards() {
 				continue
 			}
 
-			// There were no unclaimed shards, let's look for a shard to steal.
-			selfClaims := len(clientClaims[k.clientID])
-			for clientID, claims := range clientClaims {
-				if clientID == k.clientID {
-					// Don't steal from ourself, we're not at that point yet.
-					continue
-				}
-
-				// This is an extremely naive "algorithm", we simply randomly
-				// iterate all other clients with shards and if any have two
-				// more shards than we do then it's fair game. Using two here
-				// so that we don't play hot potatoes with an odd shard.
-				if len(claims) > (selfClaims + 1) {
-					randomShard := claims[(rand.Int() % len(claims))].ShardID
-					k.log.Debugf(
-						"Attempting to steal stream '%v' shard '%v' from client '%v' as client '%v'\n",
-						streamID, randomShard, clientID, k.clientID,
-					)
-
-					sequence, err := k.checkpointer.Claim(k.ctx, streamID, randomShard, clientID)
-					if err != nil {
-						if k.ctx.Err() != nil {
-							return
-						}
-						if !errors.Is(err, ErrLeaseNotAcquired) {
-							k.log.Errorf("Failed to steal shard '%v': %v\n", randomShard, err)
-						}
-						k.log.Debugf(
-							"Aborting theft of stream '%v' shard '%v' from client '%v' as client '%v'\n",
-							streamID, randomShard, clientID, k.clientID,
-						)
-						continue
-					}
-
-					k.log.Debugf(
-						"Successfully stole stream '%v' shard '%v' from client '%v' as client '%v'\n",
-						streamID, randomShard, clientID, k.clientID,
-					)
-					wg.Add(1)
-					if err = k.runConsumer(&wg, streamID, randomShard, sequence); err != nil {
-						k.log.Errorf("Failed to start consumer: %v\n", err)
-					} else {
-						// If we successfully stole the shard then that's enough
-						// for now.
-						break
-					}
-				}
+			// Note: Stealing only happens if all shards have already been claimed
+			if _, err = k.stealShards(&wg, streamID, clientClaims); err!=nil {
```

Comment:
I think you need to to format this file using goimports

Reply:
gotcha 👍